net/dns/resolver/tsdns.go

```diff
@@ -246,6 +249,15 @@ func (r *Resolver) SetConfig(cfg Config) error {
 	r.localDomains = cfg.LocalDomains
 	r.hostToIP = cfg.Hosts
 	r.ipToHost = reverse
+
+	// Extract wildcard hosts
+	keys := make(map[dnsname.FQDN][]netip.Addr)
```

Comment:
In the common case we won't need this map so I'd just declared it here with its nil zero value and not use make.  You can use mak.Set in the loop to lazily set it.